ffi/diplomat/src/lib.rs

```diff
@@ -44,40 +44,68 @@ extern crate std as rust_std;
 
 extern crate alloc;
 
+#[cfg(feature = "icu_properties")]
 pub mod bidi;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod calendar;
+#[cfg(feature = "icu_collator")]
 pub mod collator;
 pub mod common;
 pub mod data_struct;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod date;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod datetime;
+#[cfg(feature = "icu_datetime")]
 pub mod datetime_formatter;
+#[cfg(feature = "icu_decimal")]
 pub mod decimal;
+#[cfg(feature = "icu_displaynames")]
 pub mod displaynames;
 pub mod errors;
 pub mod fallbacker;
+#[cfg(feature = "icu_decimal")]
 pub mod fixed_decimal;
+#[cfg(feature = "icu_list")]
 pub mod list;
 pub mod locale;
+#[cfg(feature = "icu_locid_transform")]
 pub mod locid_transform;
 pub mod logging;
+#[cfg(feature = "icu_normalizer")]
 pub mod normalizer;
+#[cfg(feature = "icu_normalizer")]
 pub mod normalizer_properties;
+#[cfg(feature = "icu_plurals")]
 pub mod pluralrules;
+#[cfg(feature = "icu_properties")]
 pub mod properties_maps;
+#[cfg(feature = "icu_properties")]
 pub mod properties_names;
+#[cfg(feature = "icu_properties")]
 pub mod properties_sets;
+#[cfg(feature = "icu_properties")]
 pub mod properties_unisets;
 pub mod provider;
+#[cfg(feature = "icu_properties")]
 pub mod script;
+#[cfg(feature = "icu_segmenter")]
 pub mod segmenter_grapheme;
+#[cfg(feature = "icu_segmenter")]
 pub mod segmenter_line;
+#[cfg(feature = "icu_segmenter")]
 pub mod segmenter_sentence;
+#[cfg(feature = "icu_segmenter")]
 pub mod segmenter_word;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod time;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod timezone;
```

Comment:
https://github.com/rust-diplomat/diplomat/pull/321

Reply:
> Do we also want to support the use case of the opposite direction? I.e. formatting but no metazone?

I think for now we should not break up modules (so icu_datetime enables both icu_calendar and icu_timezone unconditionally) and we can improve this later if necessary. For the time being we don't have clients who only need a subset of icu_datetime.

Reply:
As an example:

- `icu_capi/icu_datetime` enables all APIs corresponding to the `icu_datetime` crate
- `icu_capi/datetime_plain` enables the APIs for `icu_datetime` for PlainDateTime formatting (no time zones)
- `icu_capi/datetime_timezone` enables the APIs for `icu_datetime` for TimeZone formatting (no calendars)

Reply:
@sffc So what I was considering was that `icu_datetime` enables timezone formatting but not metazone calculation, and `icu_timezone` enables metazone calculation but not formatting. Both must enable the date/time/timezone _types_ since they're all intertwined.


> so icu_datetime enables both icu_calendar and icu_timezone unconditionally

This is ambiguous, since there's a difference between deps and functionality. We _have_ to enable these deps anyway.

Reply:
Yep; that aligns with my assumptions.

Reply:
I realized after I wrote the above comment that if we wanted to avoid the icu_timezone _dependency_ if you are only formatting PlainDateTimes, we'd need to add that as a feature to the icu_datetime crate. We should not do that now (and it remains unclear if we need to in the future)

Reply:
Yeah I'm not going that far right now